Assets/__Scripts/MapEditor/Grid/Collections/EventsContainer.cs

```diff
@@ -18,13 +19,33 @@ public bool RingPropagationEditing
         set
         {
             ringPropagationEditing = value;
-            ringPropagationLabels.SetActive(value);
-            eventGridLabels.SetActive(!value);
+            PlatformDescriptor pD = FindObjectOfType<PlatformDescriptor>();
```

Comment:
Making use of the previous `LoadInitialMap.PlatformLoadedEvent`, you can store a local reference of the `PlatformDescriptor` and use that instead of calling the expensive `FindObjectOfType<>()` every time Ring Propagation is changed.